core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -223,6 +224,69 @@ private static class AliasReplacementShuttle extends SqlShuttle {
     }
   }
 
+  /**
+   * Wraps a nested join Result into a subquery with a new alias.
+   * Required for dialects like ClickHouse that don't support nested JOIN syntax.
+   *
+   * @param input the Result from the nested join subtree
+   * @param outerAlias the alias to assign to the wrapped subquery
+   * @return a new Result with the wrapped SQL node
+   */
+  protected Result wrapNestedJoin(Result input, RelNode inputRel, String outerAlias) {
+    // Get the original SQL node from the input Result
+    final SqlNode original = input.asSelect();
+    // Generate inner alias (different from outer)
+    String innerAlias = "t" + inputRel.getId();
+    // Wrap: original → (SELECT * FROM (original) AS newAlias)
+    SqlNode wrapped = wrapAsSelectStar(original, innerAlias);
+
+    Map<String, RelDataType> newAliases = new LinkedHashMap<>();
```

Comment:
Is there any consideration for sorting efficiency when using LinkedHashMap here?